src/EmployeesAdministrationApp.java

```diff
@@ -0,0 +1,148 @@
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.json.simple.JSONObject;
+
+public class EmployeesAdministrationApp {
+
+	public static void main(String[] args) {
+
+		switch (chooseAnOptionFromMainMenu()) {
+		case 1:
+			addNewEmployee();
+			break;
+		case 2:
+			viewAllEmployees();
+			break;
+		case 3:
+			searchEmployee();
+			break;
+		case 4:
+			deleteEmployee();
+			break;
+		case 5:
+			updateEmployee();
+			break;
+		case 6:
+			convertEmployee();
+			break;
+		case 7:
+			System.out.println("Bye Bye!");
+			break;
+		}
+	}
+
+	private static void addNewEmployee() {
+		System.out.println("Please select the type of the new employee:"
+				+ "\n1. QA Employee"
+				+ "\n2. Developer Employee"
+				+ "\n3. QA Team Lead Employee " + "\n4. Developer Team Lead Employee" + "\n5. Project Manager");
+			Scanner scan = new Scanner(System.in);
+			int typeOfTheEmployee = scan.nextInt();
+
+			switch (typeOfTheEmployee) {
+			case 1: {
+				QAEmployee qa = new QAEmployee();
+				System.out.println("Enter the QA employee data");
+				qa.addDetailsForEmployee();
+				writeJson(qa);
+				break;
+			}
+			case 2: {
+				DeveloperEmployee dev = new DeveloperEmployee();
+				System.out.println("Enter the Developer employee data");
+				dev.addDetailsForEmployee();
+				writeJson(dev);
+				break;
+			}
+			case 3: {
+				QATeamLead qaTeamLead = new QATeamLead();
+				System.out.println("Enter the QA Team Lead employee data");
+				qaTeamLead.addDetailsForEmployee();
+				writeJson(qaTeamLead);
+				break;
+			}
+			case 4: {
+				DeveloperTeamLead developerTeamLead = new DeveloperTeamLead();
+				System.out.println("Enter the Developer Team Lead employee data");
+				developerTeamLead.addDetailsForEmployee();
+				writeJson(developerTeamLead);
+				break;
+			}
+			case 5: {
+				ProjectManager projectManager = new ProjectManager();
+				System.out.println("Enter the Project Manager employee data");
+				projectManager.addDetailsForEmployee();
+				writeJson(projectManager);
+				break;
+			}
+			}
+			scan.close();
+	}
+
+	private static void writeJson(Object obj) {
+		Method[] methods = obj.getClass().getMethods();
+		HashMap<String, Object> employeeDetails = new HashMap<String, Object>();
```

Comment:
use Map when declaring

Reply:
Map<String, Object> employeeDetails = new HashMap<String, Object>();

Reply:
I added the correct declaration.